src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncIteratorTests.cs

```diff
@@ -22,25 +22,581 @@ internal enum Instruction
     public class CodeGenAsyncIteratorTests : EmitMetadataTestBase
     {
         // PROTOTYPE(async-streams)
-        // Test missing types/members
+        // Test missing remaining types/members once BCL APIs are finalized (MRVTSL, IStrongBox, IValueTaskSource)
+        // test missing AsyncTaskMethodBuilder<T> or missing members Create(), Task, ...
         // Test with yield or await in try/catch/finally
         // More tests with exception thrown
         // There is a case in GetIteratorElementType with IsDirectlyInIterator that relates to speculation, needs testing
         // yield break disallowed in finally and top-level script (see BindYieldBreakStatement); same for yield return (see BindYieldReturnStatement)
-        // test local function
         // binding for yield return (BindYieldReturnStatement) validates escape rules, needs testing
         // test yield in async lambda (still error)
         // test exception handling (should capture and return the exception via the promise)
         // test IAsyncEnumerable<U> M<U>() ...
-        // test missing AsyncTaskMethodBuilder<T> or missing members Create(), Task, ...
         // test with IAsyncEnumerable<dynamic>
         // other tests with dynamic?
         // test should cover both case with AwaitOnCompleted and AwaitUnsafeOnCompleted
         // test `async IAsyncEnumerable<int> M() { return TaskLike(); }`
         // Can we avoid making IAsyncEnumerable<T> special from the start? Making mark it with an attribute like we did for task-like?
-        // Test normal break in async-iterator
-        // WaitForNextAsync is resilient to be called out of turn. Test that.
-        // Test a plain return statement
+        // Do some manual validation on debugging scenarios, including with exceptions (thrown after yield and after await).
+
+        [Fact]
+        public void MissingType_IAsyncIEnumerable()
+        {
+            string source = @"
+class C
+{
+    async System.Collections.Generic.IAsyncEnumerable<int> M()
+    {
+        await System.Threading.Tasks.Task.CompletedTask;
+        yield return 3;
+    }
+}
+namespace System.Collections.Generic
+{
+    public interface IAsyncEnumerator<out T> : System.IAsyncDisposable
+    {
+        System.Threading.Tasks.ValueTask<bool> WaitForNextAsync();
+        T TryGetNext(out bool success);
+    }
+}
+namespace System
+{
+    public interface IAsyncDisposable
+    {
+        System.Threading.Tasks.ValueTask DisposeAsync();
+    }
+}
```

Comment:
Are these used?

Reply:
They are not used, but I wanted to only have one thing missing where possible.

Reply:
I've refactored those tests to avoid repetition, which should address this comment.